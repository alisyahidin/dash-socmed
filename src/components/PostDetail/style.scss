.comment-icon {
  color: #666;
}