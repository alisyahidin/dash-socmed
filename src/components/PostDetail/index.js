import React from 'react'
import { Link } from 'react-router-dom'
import { Modal } from 'react-bootstrap'
import { FaRegComment } from 'react-icons/fa'

import Comment from '../Comment/'
import CommentForm from '../Comment/Form/'
import './style.scss'

const PostDetail = ({open, toggleOpen}) => (
  <Modal size="md" show={open} onHide={() => toggleOpen(false)}>
    <Modal.Header closeButton>
      <Modal.Title>
        <h2>Lorem Ipsum</h2>
        <Link to="/profile" className="author d-flex align-items-center mt-sm-2">
          <h5>
            Ali <span className="author__username">@ali</span>
          </h5>
        </Link>
      </Modal.Title>
    </Modal.Header>
    <Modal.Body className="px-4">
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio,
        tempore error autem atque quas corrupti, cupiditate. Earum voluptate,
        error, ipsam itaque, repellat commodi aliquam id veniam neque explicabo
        iste sequi.
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio,
        tempore error autem atque quas corrupti, cupiditate. Earum voluptate,
        error, ipsam itaque, repellat commodi aliquam id veniam neque explicabo
        iste sequi.
      </p>

      <span className="comment-icon mt-sm-2 d-flex align-items-center">
        <FaRegComment className="mr-sm-1" /> 5
      </span>
    </Modal.Body>
    <Modal.Footer className="flex-column">
      <CommentForm />
      <hr className="w-100" />
      <Comment />
      <Comment />
      <Comment />
    </Modal.Footer>
  </Modal>
)

export default PostDetail