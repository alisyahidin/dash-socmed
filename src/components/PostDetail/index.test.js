import React from 'react'
import { shallow, mount } from 'enzyme'
import PostDetail from './index'

it('render <PostDetail /> correctly', () => {
  const component = shallow(<PostDetail />)

  expect(component).toMatchSnapshot()
})