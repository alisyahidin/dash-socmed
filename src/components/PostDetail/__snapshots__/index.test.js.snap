// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <PostDetail /> correctly 1`] = `
<ForwardRef(Bootstrap(Modal))
  onHide={[Function]}
  size="md"
>
  <ForwardRef(Bootstrap(ModalHeader))
    closeButton={true}
  >
    <ForwardRef(Bootstrap(ModalTitle))>
      <h2>
        Lorem Ipsum
      </h2>
      <Link
        className="author d-flex align-items-center mt-sm-2"
        replace={false}
        to="/profile"
      >
        <h5>
          Ali 
          <span
            className="author__username"
          >
            @ali
          </span>
        </h5>
      </Link>
    </ForwardRef(Bootstrap(ModalTitle))>
  </ForwardRef(Bootstrap(ModalHeader))>
  <ForwardRef(Bootstrap(ModalBody))
    className="px-4"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio, tempore error autem atque quas corrupti, cupiditate. Earum voluptate, error, ipsam itaque, repellat commodi aliquam id veniam neque explicabo iste sequi. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio, tempore error autem atque quas corrupti, cupiditate. Earum voluptate, error, ipsam itaque, repellat commodi aliquam id veniam neque explicabo iste sequi.
    </p>
    <span
      className="comment-icon mt-sm-2 d-flex align-items-center"
    >
      <FaRegComment
        className="mr-sm-1"
      />
       5
    </span>
  </ForwardRef(Bootstrap(ModalBody))>
  <ForwardRef(Bootstrap(ModalFooter))
    className="flex-column"
  >
    <CommentForm />
    <hr
      className="w-100"
    />
    <PostDetail />
    <PostDetail />
    <PostDetail />
  </ForwardRef(Bootstrap(ModalFooter))>
</ForwardRef(Bootstrap(Modal))>
`;
